tests/other/metadce/mem_no_main_O3_STANDALONE_WASM.imports

```diff
@@ -1 +1,3 @@
-
+args_get
+args_sizes_get
```

Comment:
this seems wrong, if there is no main, we shouldn't use args_get?

Reply:
So far I've not found a way to handle the following scenario:

1. You don't have a main
2. You are using STANDALONE_WASM mode
3. You want to avoid all the costs associationed with _start and argv etc.
4. You won't want to pass any extra linker flags to signal that you don't have a main

I'm kind of OK with dropping the transparent handling of this use case.  Because (a) WASI doesn't yet support programs without main (b) it seems reasonable that if you really care about this configuration we can expect you to pass something like `-no-entry`, `-nostdlib`, `-s EXPORT_FUNCTIONS` without main included, to the linker.

So I propose the we main linking without a main a linker error in WASM_STANDALONE mode and ask people to explicitly opt into this using EXPORTED_FUNCTIONS with main not present.. does that seems reasonable?

Reply:
What concerns me is that the standalone case without main is basically the standard "plugin" case. So I actually think it's common, more common than running wasi in the shell.

>  ask people to explicitly opt into this using EXPORTED_FUNCTIONS with main not present.. does that seems reasonable?

That seems reasonable - it looks like main was the one exception to this rule actually, as we already error on anything else in the list that isn't actually present.

Reply:
Ah yes, I think I'm proposing that (for WASM_STANDALONE at least) we remove that exception and show an error such as `main symbol not defined, if that is intentional please build with flag XXX to build a program without a main`

Reply:
Sounds good. Can the flag just be not having `_main` in EXPORTED_FUNCTIONS?

Reply:
Yes